doc/keynames/README.md

```diff
@@ -0,0 +1,477 @@
+  Currently, there is no way of knowing which filesystem directory a Key with Namespace "Dir" is scoped to.
+  Elektra only uses Keys scoped to the Current Working Directory.
+- Similarly, Keys with Namespace **"User"** are scoped to a single **User**.
+  They will (normally) be stored somewhere within the User's home directory.
+  Again, there is no way of knowing which user a Key is scoped to and Elektra only uses Keys scoped to the current user, i.e. the one that executed the application.
+- The Namespace **"System"** is what makes Elektra global.
+  Keys with Namespace "System" are the same for all users of the system, independent of context.
+  They are stored in system level directory.
+- Keys with Namespace "Default" are special.
+  While you could create them manually, you normally don't want to.
+  It is used for Keys with default values for Namespace Resolution (explained below).
+  Keys with this Namespace are also in-memory only.
+- Then there is **"Spec"**.
+  This Namespace, like "Meta", is separate from the rest.
+  We use it for Keys that are part of a specification used to describe other Keys.
+  The Metadata of every Key with Namespace "Spec" describes a specification for the Keys that have the same Key Name Part but a different Namespace (except "Meta").
+  So the Namespace "Spec" has a closer relation to the others than "Meta".
+- Now just the Namespace **"Cascading"** remains.
+  This Namespace is used for Namespace Resolution (see below).
+  It is the one that applications and end-users will use most commonly when interacting with Elektra.
+  Keys with Namespace "Cascading" are never stored.
+  Not on disk and ideally also not in a `KeySet`.
+
+There is also a certain ranking between the Namespaces "Proc", "Dir", "User", "System" and "Default".
+Namely, that they override each other in exactly this order.
+Given two Key Names with identical Key Name Parts, but one with Namespace "Dir" and one with Namespace "User", the one with Namespace "Dir" should be considered more specific and should be preferred.
+
+A special feature of Elektra is _Namespace Resolution_ (sometimes just Name Resolution, or Key Name Resolution).
+Namespace Resolution is the process of finding an appropriate Namespace for a Key based on a Key Name with Namespace "Cascading".
+It is most commonly used, when you to find which Key in the KDB should be used, based on a series of Key Name Parts.
+
+To resolve the namespace, we just look at each of the Namespaces in the ranking defined above.
+We then use the first Namespace where the Key actually exists.
+Namespace Resolution is performed, when `ksLookup`/`ksLookupByName` is called with a Key Name with Namespace "Cascading" [[2]](#footnote-2).
+<a id="ref-footnote-2"></a>
+This is also done, if you call `kdb get` or `kdb set` with a Key Name with Namespace "Cascading".
+
+## 2. Escaped Names
+
+The standard way to represent a Key Name in Elektra is this:
```

Comment:
```suggestion
We start with an example how to deconstruct following key name:
```

Reply:
I don't think it makes sense to say "we deconstruct a key name", when until this point the document defines a key name as "a namespace, plus a series of key name parts".

Maybe this would be better:

> The most common way to write down the key in the namespace "System" which has the key name parts `elektra`, `version` and `info` is this:

Then we have a transition from the definition above to the representation that we want to introduce.